modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -352,11 +350,20 @@ public CommerceOrder addCommerceOrder(
 				commerceChannel, commerceOrder);
 		}
 
-		_setAccountDefaultPaymentTerm(commerceChannel, commerceOrder, user);
-
 		commerceOrder.setName(name);
 		commerceOrder.setPurchaseOrderNumber(purchaseOrderNumber);
-		commerceOrder.setShippingOptionName(shippingOptionName);
+
+		if ((commerceShippingMethodId > 0) && (shippingOptionName != null)) {
+			commerceOrder.setCommerceShippingMethodId(commerceShippingMethodId);
+			commerceOrder.setShippingOptionName(shippingOptionName);
+		}
+		else {
+			_setAccountDefaultShippingOption(
+				commerceChannel, commerceOrder, user);
+		}
+
+		_setAccountDefaultDeliveryTerm(commerceChannel, commerceOrder, user);
+		_setAccountDefaultPaymentTerm(commerceChannel, commerceOrder, user);
 
```

Comment:
Same situation here, `_setAccountDefaultDeliveryTerm` and `_setAccountDefaultPaymentTerm` need to be after `setShippingOptionName`